src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -922,25 +916,31 @@ MulticopterPositionControl::reset_alt_sp()
 void
 MulticopterPositionControl::limit_altitude()
 {
-	/* in altitude control, limit setpoint */
-	if (_run_alt_control && _pos_sp(2) <= -_vehicle_land_detected.alt_max) {
-		_pos_sp(2) = -_vehicle_land_detected.alt_max;
-		return;
-	}
+	if (_vehicle_land_detected.alt_max > 0.0f) {
```

Comment:
this part gets replaced with https://github.com/PX4/Firmware/pull/8048/files#diff-fc77c3ef569029d45764664c75d4b0c1R925